libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp

```diff
@@ -215,14 +253,55 @@ sidechain_net_handler_bitcoin::sidechain_net_handler_bitcoin(std::shared_ptr<gra
 sidechain_net_handler_bitcoin::~sidechain_net_handler_bitcoin() {
 }
 
+son_wallet_update_operation sidechain_net_handler_bitcoin::recreate_primary_wallet() {
+   const auto& idx_swi = database.get_index_type<son_wallet_index>().indices().get<by_id>();
+   auto obj = idx_swi.rbegin();
+   if (obj != idx_swi.rend()) {
+
+      if ((obj->addresses.find(sidechain_type::bitcoin) == obj->addresses.end()) ||
+          (obj->addresses.at(sidechain_type::bitcoin).empty())) {
+         auto active_sons = database.get_global_properties().active_sons;
+         vector<string> son_pubkeys_bitcoin;
+         for ( const son_info& si : active_sons ) {
+            son_pubkeys_bitcoin.push_back(si.sidechain_public_keys.at(sidechain_type::bitcoin));
+         }
+         string reply_str = create_multisignature_wallet(son_pubkeys_bitcoin);
+
+         ilog(reply_str);
+
+         std::stringstream ss(reply_str);
+         boost::property_tree::ptree pt;
+         boost::property_tree::read_json( ss, pt );
+         if( pt.count( "error" ) && pt.get_child( "error" ).empty() ) {
+            ilog(__FUNCTION__);
+
+            son_wallet_update_operation op;
+            op.payer = database.get_global_properties().parameters.get_son_btc_account_id();
```

Comment:
If the consensus account is used as the paying account at least 2/3 SONs have to sign for the transaction to go through. So it won't be possible with just one SON pushing the transaction, this transaction has to be put into a proposal and get all the SON signatures. And each SON has to detect the created proposal for SON operations and send the proposal approval. Please refer to peerplays_sidechain_plugin_impl::on_objects_new and peerplays_sidechain_plugin_impl::create_son_down_proposals.